code/modules/mob/mob_helpers.dm

```diff
@@ -398,4 +398,59 @@ mob/proc/get_standard_bodytemperature()
 	if(client?.player_details)
 		client.player_details.logging[smessage_type] += timestamped_message
 
-	return ..()
\ No newline at end of file
+	return ..()
+
+
+/proc/notify_ghost(mob/dead/observer/O, message, ghost_sound = null, enter_link = null, enter_text = null, atom/source = null, mutable_appearance/alert_overlay = null, action = NOTIFY_JUMP, flashwindow = TRUE, ignore_mapload = TRUE, ignore_key, header = null, notify_volume = 100, extra_large = FALSE) //Easy notification of a single ghosts.
+	if(ignore_mapload && SSatoms.initialized != INITIALIZATION_INNEW_REGULAR)	//don't notify for objects created during a map load
+		return
+	if(!O.client)
+		return
+	var/track_link
+	if (source && action == NOTIFY_ORBIT)
+		track_link = " <a href='byond://?src=[REF(O)];track=[REF(source)]'>(Follow)</a>"
+	if (source && action == NOTIFY_JUMP)
+		var/turf/T = get_turf(source)
+		track_link = " <a href='byond://?src=[REF(O)];jump=1;x=[T.x];y=[T.y];z=[T.z]'>(Jump)</a>"
+	var/full_enter_link
+	if (enter_link)
+		full_enter_link = "<a href='byond://?src=[REF(O)];[enter_link]'>[(enter_text) ? "[enter_text]" : "(Claim)"]</a>"
+	to_chat(O, "[(extra_large) ? "<br><hr>" : ""]<span class='deadsay'>[message][(enter_link) ? " [full_enter_link]" : ""][track_link]</span>[(extra_large) ? "<hr><br>" : ""]")
+	if(ghost_sound)
+		SEND_SOUND(O, sound(ghost_sound, volume = notify_volume))
+	if(flashwindow)
+		window_flash(O.client)
+
+	if(!source)
+		return
+
+	var/obj/screen/alert/notify_action/A = O.throw_alert("[REF(source)]_notify_action", /obj/screen/alert/notify_action)
+	if(!A)
+		return
+	if (header)
+		A.name = header
+	A.desc = message
+	A.action = action
+	A.target = source
+	if(!alert_overlay)
+		alert_overlay = new(source)
+	alert_overlay.layer = FLOAT_LAYER
+	alert_overlay.plane = FLOAT_PLANE
+
+	// Extra processing for queen notifications
+	if (isxenoqueen(source))
```

Comment:
And instead, here you could make use of `Height()` and `Width()` to find out how many pixels the image has.
Take the larger value. If it's higher than 32, scale the whole thing down proportionally. Else, take the lower value. If smaller than 32, or whatever other value you want, increase it.